mast_aladin/adapters/imviz_sync_adapter.py

```diff
@@ -1,22 +1,48 @@
 from .viewer_sync_adapter import ViewerSyncAdapter
+import warnings
 
 
 class ImvizSyncAdapter(ViewerSyncAdapter):
     def __init__(self, viewer=None):
-        # todo: assert the type of the viewer is jdaviz/imviz
         from jdaviz.configs.imviz.helper import _current_app
         self.app = viewer if viewer else _current_app
         self.viewer = self.app.default_viewer
-        self.aid = self.viewer._obj.aid
+        self.aid = self.viewer._obj.glue_viewer.aid
 
-    def sync_to(self, sync_viewer):
-        self.aid.set_viewport(**sync_viewer.aid.get_viewport())
+    def sync_to(self, sync_viewer, aspects):
+        source_viewport = sync_viewer.aid.get_viewport(sky_or_pixel="sky")
+
+        new_viewport = self.aid.get_viewport(sky_or_pixel="sky").copy()
+        if "center" in aspects:
+            new_viewport["center"] = source_viewport["center"]
+        if "fov" in aspects:
+            new_viewport["fov"] = source_viewport["fov"]
+        if "rotation" in aspects:
+            new_viewport["rotation"] = source_viewport["rotation"]
+
+        self.aid.set_viewport(**new_viewport)
 
     def add_callback(self, func):
-        self.viewer._obj.state.add_callback('zoom_radius', func)
+        state = self.viewer._obj.glue_viewer.state
+        for name in ['zoom_radius', 'x_min', 'x_max', 'y_min', 'y_max']:
+            state.add_callback(name, func)
+
+        try:
+            self.app.plugins['Orientation'].rotation_angle.add_callback(func)
+        except Exception as e:
+            warnings.warn(f"Failed to add callback for rotation: {e}")
 
     def remove_callback(self, func):
-        self.viewer._obj.state.remove_callback('zoom_radius', func)
+        state = self.viewer._obj.glue_viewer.state
+        for name in ['zoom_radius', 'x_min', 'x_max', 'y_min', 'y_max']:
+            try:
+                state.remove_callback(name, func)
+            except Exception as e:
+                warnings.warn(f"Failed to remove callback {name}: {e}")
```

Comment:
I think we want to `pass` on an exception here similar to `aladin_sync_adapter` as the user shouldn't have to know anything about this imo

Reply:
My philosophy is that we want to warn only when its something that isn't breaking anything directly, and the user has the capacity to fix the issue on their end. If they can't do anything about it, its just an annoyance to them IMO. This is the kind of thing that would be really good as a debug log however so that we can turn it on, but users never have to see it! 

Here is a [quick tutorial](https://docs.python.org/3.6/howto/logging.html#logging-basic-tutorial) on the logging library in python. It would require a bit of a refactor of the package to enable this, but it would help a lot with these sorts of questions